monitoring/serf.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2016 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package agent
+
+import (
+	"sync"
+
+	"github.com/gravitational/trace"
+	serf "github.com/hashicorp/serf/client"
+)
+
+// serfClient is the minimal interface to the serf cluster.
+// It enables mocking access to serf network in tests.
+type serfClient interface {
```

Comment:
Yeah, isn't this file basically a verbatim copy of `agent/serf.go`? Why not import it from there?

Reply:
same answer, I gave above, but:
I was thinking about that but in order to do that, shouldn't I change `serfClient` to be `SerfClient` thus needing a refactoring in all the `agent` package where that's used?

Is that okay?

Reply:
Sure, I see nothing wrong with exporting serf client. Moreover, I bet your refactoring will be very short - just capitalize "s" in a couple of spots :)

Reply:
I didn't mind changing variables in my code but I was more concerned about changing it everywhere in `satellite`... if that's okay I can definitely do it. Just didn't want to assume it was a good thing without @a-palchikov and @r0mant approval.
I'll go ahead and do it then.

Reply:
Done too.